components/camel-tahu/src/main/java/org/apache/camel/component/tahu/TahuEdgeMetricHandler.java

```diff
@@ -0,0 +1,257 @@
+        return Optional.ofNullable(descriptorMetricMap.get(sd)).map(SparkplugBPayloadMap::getMetrics).orElse(List.of());
+    }
+
+    SparkplugBPayloadMap getDescriptorMetricMap(SparkplugDescriptor sd) {
+        return descriptorMetricMap.get(sd);
+    }
+
+    void updateCachedMetrics(SparkplugDescriptor sd, SparkplugBPayload payload) {
+        Optional.ofNullable(descriptorMetricMap.get(sd)).ifPresent(cachedMetrics -> {
+            payload.getMetrics().stream().forEach(payloadMetric -> {
+                cachedMetrics.updateMetricValue(payloadMetric.getName(), payloadMetric, null);
+            });
+        });
+    }
+
+    long getCurrentBirthBdSeq() {
+        LOG.trace(loggingMarker, "getCurrentBirthBdSeq() : {}", currentBirthBdSeq);
+        return currentBirthBdSeq;
+    }
+
+    long getCurrentDeathBdSeq() {
+        LOG.trace(loggingMarker, "getCurrentDeathBdSeq() : {}", currentDeathBdSeq);
+        return currentDeathBdSeq;
+    }
+
+    List<Metric> processCMDMetrics(SparkplugBPayload payload, SparkplugDescriptor cmdDescriptor) {
+        List<Metric> receivedMetrics = payload.getMetrics();
+        if (receivedMetrics == null || receivedMetrics.isEmpty()) {
+            return List.of();
+        }
+
+        // Look for a METRIC_NODE_REBIRTH received metric with True value
+        if (!cmdDescriptor.isDeviceDescriptor()) {
+            Map<Boolean, List<Metric>> groupedMetrics = receivedMetrics.stream()
+                    .collect(Collectors.groupingBy(m -> (Boolean) SparkplugMeta.METRIC_NODE_REBIRTH.equals(m.getName())
+                            && m.getDataType() == MetricDataType.Boolean && (Boolean) m.getValue()));
+
+            if (groupedMetrics.containsKey(Boolean.TRUE) && !groupedMetrics.get(Boolean.TRUE).isEmpty()) {
+                client.handleRebirthRequest(true);
+            }
+
+            receivedMetrics = groupedMetrics.get(Boolean.FALSE);
+        }
+
+        final SparkplugBPayloadMap cachedMetrics = descriptorMetricMap.get(cmdDescriptor);
+        if (cachedMetrics == null) {
+            return List.of();
+        }
+
+        List<Metric> responseMetrics = receivedMetrics.stream().map(m -> {
+            Metric cachedMetric = cachedMetrics.getMetric(m.getName());
+
+            if (cachedMetric == null) {
+                LOG.warn(loggingMarker, "Received CMD request for {} metric {} not in configured metrics - skipping",
+                        cmdDescriptor, m.getName());
+                return null;
+            }
+
+            try {
+                Metric responseMetric = new Metric(cachedMetric);
+
+                responseMetric.setHistorical(true);
+
+                return responseMetric;
+            } catch (Exception e) {
+                LOG.warn(loggingMarker,
+                        "Exception caught copying metric handling CMD request for {} metric {} - skipping",
+                        cmdDescriptor, m.getName());
+                return null;
+            }
```

Comment:
(advisory) Maybe move this block to a separate method. It makes the code easier to read/maintain.

Reply:
Also consider it for other blocks like this.